Custom Component

sanandnarayan opened the pull request.

ashwinKumar0505 committed:
feat(sidebar): create custom components in the sidebar

ashwinKumar0505 committed:
improvement(custom components): bug in custom components

ashwinKumar0505 committed:
improvement(components): make the code reusable

ashwinKumar0505 committed:
removed console logs

ashwinKumar0505 committed:
improvement(custom components): handling custom components and code improvement

components can be dragged into the custom components and that change is affected in every instances
of the particular custom componenet. Code improvement in showing the custom components and nesting
of custom components inside custom components.

ashwinKumar0505 commented:
Yet to be done on this branch

- [x] state handling for custom components when a component or custom component in the editor is moved into the another custom component

- [x] handling Deletion of components inside custom components.

- [x]  state handling for custom components when a prop for the custom component changes.

- [x] Code Refactor.

ashwinKumar0505 committed:
improvement(custom components): state handling of custom components

state handling for custom components when a component or custom component in the editor is moved
into the another custom component

ashwinKumar0505 commented:
Just as of now. If a new component is added to a particular custom component , It will be affected in all the instances of the custom component. Needs to be edited later based upon the needs of the project.

ashwinKumar0505 committed:
improvement(custom components): handling Deletion of components inside custom components

ashwinKumar0505 committed:
fix(custom components): bug in handling the custom components state

This will fix the updation of all the instances custom component when a component is moved into the
particular custom component.

ashwinKumar0505 committed:
fix(custom components): state handling for custom components

ashwinKumar0505 committed:
improvement(custom components): state handling for the styles of custom components

state handling for custom components when a prop for the custom component changes.

ashwinKumar0505 committed:
fix(code preview): shows wrong code preview for custom components

ashwinKumar0505 committed:
refactor(custom components): code is refactored

ashwinKumar0505 committed:
fix(custom components): fix : wrong code exported to codesandbox

ashwinKumar0505 committed:
fix(custom components): clear option will now clears the custom components also

ashwinKumar0505 committed:
fix: bug in deletion of custom components

ashwinKumar0505 committed:
improvement(custom components): added a way to import and save custom components

ashwinKumar0505 commented:
Right now , custom components can be imported and used in the project where it was created. Next step is to make the custom components to be imported and use in any projects.

ashwinKumar0505 committed:
fix(deletion): deleting component that contains custom component deletes children of custom comp

ashwinKumar0505 committed:
refactor(custom components): the code is refactored

ashwinKumar0505 committed:
fix(adding custom component): a custom component can not be added to its same instance

ashwinKumar0505 committed:
importing custom components now replace the custom components in state
